### PR TITLE
Adding enhancement to route for v2 prov

### PR DESCRIPTION
**Pull request by MbolotSuse**
Relies on rancher/steve#56 to go in first.

### CODE REVIEW

**Comment by MbolotSuse** [comment]
Build has been cancelled for now until the PR for steve goes in. This will require steve to be updated to take advantages of the changes made there.

File: pkg/api/steve/proxy/proxy.go [hunk 1]
```diff
@@ -87,8 +87,10 @@ func routeToShellProxy(key, value string, localSupport bool, localCluster http.H
 			if localSupport {
 				authed := proxyHandler.userCanAccessCluster(r, cluster)
 				if !authed {
-					rw.WriteHeader(http.StatusUnauthorized)
-					return
+					if r.Context().Value(auth.CattleAuthFailed) == nil {
```
Comment (cbron) [inline pkg/api/steve/proxy/proxy.go:90]: nit here and below: 
```suggestion
					if r.Context().Value(auth.CattleAuthFailed) != "true" {
```

File: pkg/api/steve/proxy/proxy.go [hunk 2]
```diff
@@ -65,7 +65,7 @@ func NewProxyMiddleware(sar v1.AuthorizationV1Interface,
 
 	mux := gmux.NewRouter()
 	mux.UseEncodedPath()
-	mux.PathPrefix("/api").HandlerFunc(proxyHandler.authLocalCluster(mux))
+	mux.PathPrefix("/api").HandlerFunc(proxyHandler.authLocalCluster(localCluster))
```
Comment (cbron) [inline pkg/api/steve/proxy/proxy.go:68]: I don't understand this well enough to say whether this is good or bad, but seems noteworthy ?

File: pkg/api/steve/proxy/proxy.go (hunk 1 above)
Comment (rmweir) [reply pkg/api/steve/proxy/proxy.go]: I agree here that we could be more explicit.

**Review by rmweir — CHANGES_REQUESTED** [review]
Besides @cbron feedback about checking for actual value instead of nil, LGTM

**Comment by MbolotSuse** [comment]
Letting the build go through - this is ready now.

**Review by cbron — APPROVED** [review]
lgtm when build passes

**Review by rmweir — APPROVED** [review]
LGTM

**Pull request merged**